validator_client/src/initialized_validators.rs

```diff
@@ -351,6 +396,84 @@ impl InitializedValidators {
         Ok(())
     }
 
+    /// Tries to decrypt the key cache.
+    ///
+    /// Returns `Ok(true)` if decryption was successful, `Ok(false)` if it couldn't get decrypted
+    /// and an error if a needed password couldn't get extracted.
+    ///
+    fn try_decrypt_key_cache(
+        &self,
+        cache: &mut KeyCache,
```

Comment:
Taking the `cache` by ownership and returning it inside `Result::Ok` would avoid the need for the `bool` and make it more difficult to accidentally use a cache that failed decryption.

Reply:
I wanted to distinguish between decryption failed because of a missing password which is a error not just for the cache but for the whole application (reading the single keystores won't help) and other reasons just concerning the cache in which case the cache can be ignored and the keystores read, but I assume this may be better designed by just adding new Error cases and handling them different.